packages/blade/src/components/Input/PhoneNumberInput/PhoneNumberInput.web.tsx

```diff
@@ -121,6 +144,19 @@ const _PhoneNumberInput: React.ForwardRefRenderFunction<BladeElementRef, PhoneNu
       .sort((a, b) => a.name.localeCompare(b.name));
   }, [allowedCountries, flags]);
 
+  const handleKeyDown = (keyboardEvent: FormInputOnKeyDownEvent): void => {
+    validateNumericInput(keyboardEvent.event);
+  };
+
+  const handlePaste = (pasteEvent: FormInputOnPasteEvent): void => {
+    const pastedText = pasteEvent.value?.clipboardData?.getData('text') ?? '';
+
+    // Only allow pastedText if it only has numeric characters
```

Comment:
I am not sure if we should stop people from pasting numbers like `+91 9999 999999`. If you copy from whatsapp or a lot of other websites, it often comes with spaces, `-`, or `+`. We can let people paste things as is or we can replace the unsupported characters before pasting and paste the numbers, spaces, hyphens, and plus, etc.

Reply:
You can check the `<input type="number" />` behaviour actually. We can replicate that.

Reply:
Yeah skipping plus and space also fine. Overall its fine if wrongly formatted number is pasted. People can remove the 91 manually. We should just avoid blocking the paste

Reply:
Checked what `<input type="number" />` does. It strips the string from all non-numeric inputs, adding the same logic here.